pkg/k8s/watchers/watcher.go

```diff
@@ -116,12 +116,6 @@ type endpointManager interface {
 	WaitForEndpointsAtPolicyRev(ctx context.Context, rev uint64) error
 }
 
-type nodeDiscoverManager interface {
-	NodeDeleted(n nodeTypes.Node)
-	NodeUpdated(n nodeTypes.Node)
-	ClusterSizeDependantInterval(baseInterval time.Duration) time.Duration
```

Comment:
If we add a new method to this interface, all of the remaining changes in this file and in pkg/k8s/watchers/cilium_network_policy.go, daemon/cmd/daemon.go and pkg/k8s/watchers/cilium_node.go can be avoided.